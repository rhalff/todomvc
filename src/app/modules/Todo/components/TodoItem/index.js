// @flow
import React, { PureComponent } from 'react'
import styles from './scss/TodoItem.scss'
import Checkbox from '../Checkbox'
import LabelEditor from '../LabelEditor'
import RemoveButton from '../RemoveButton'
import type { Todo } from '../../types/todos'

type Props = {
  item: Todo,
  onRemove: (item: Todo) => void,
  onSave: (text: string, item: Todo) => void,
  onToggleComplete: (item: Todo) => void
}

class TodoItem extends PureComponent<Props> {
  onRemove = () => {
    this.props.onRemove(this.props.item)
  }

  onToggleComplete = () => {
    this.props.onToggleComplete(this.props.item)
  }

  onSave = (text: string) => {
    this.props.onSave(text, this.props.item)
  }

  render() {
    const { item } = this.props

    return (
      <div className={item.completed ? styles.completed : styles.active}>
        <Checkbox checked={item.completed} onChange={this.onToggleComplete} />
        <LabelEditor text={item.text} onSave={this.onSave} />
        <RemoveButton onClick={this.onRemove} />
      </div>
    )
  }
}

export default TodoItem
