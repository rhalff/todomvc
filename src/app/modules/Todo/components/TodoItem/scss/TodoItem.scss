@import 'colors';

.todoItem {
  display: flex;
  justify-content: space-between;
  align-items: center;

  button {
    display: none;
  }

  > {
    :first-child {
      flex: 0 0 auto;
    }

    :last-child {
      position: absolute;
      right: 0;
    }

    :nth-child(2) {
      flex: 1 1 auto;
    }
  }

  &:hover {
    button {
      cursor: pointer;
      display: block;
    }
  }
}

.active {
  composes: todoItem;
}

.completed {
  composes: todoItem;

  color: $todoItemColor;
  > div {
    text-decoration: line-through;
  }
}
