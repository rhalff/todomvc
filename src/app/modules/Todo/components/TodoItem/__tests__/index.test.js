// @flow
import React from 'react'
import TodoItem from '../'
import renderer from 'react-test-renderer'

const todo = { id: 1, text: 'Test Todo' }

test('Can show Todo Item', () => {
  const component = renderer.create(<TodoItem onClick={() => {}} item={todo} />)

  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
