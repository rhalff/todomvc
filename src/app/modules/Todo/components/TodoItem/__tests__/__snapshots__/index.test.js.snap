// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Can show Todo Item 1`] = `
<div
  className="active"
>
  <input
    checked={undefined}
    className="unchecked"
    onChange={[Function]}
    type="checkbox"
  />
  <div
    className="labelEditor"
  >
    <div
      contentEditable={false}
      onDoubleClick={[Function]}
      onKeyDown={[Function]}
    >
      Test Todo
    </div>
  </div>
  <button
    className="removeButton"
    onClick={[Function]}
  >
    Ã—
  </button>
</div>
`;
